live-build/misc/migration-scripts/dx_apply

```diff
@@ -269,8 +275,25 @@ MAIN_MENU_FICL=(
 } >>/boot/menu.rc.local ||
 	die "failed to update /boot/menu.rc.local"
 
-umount "$TMP_ROOT/var/delphix" ||
-	die "couldn't unmount linux directory $TMP_ROOT/var/delphix"
+#
+# Note that the new /var/delphix should not contain any useful data as it will
+# be replaced by a clone of the current /var/delphix dataset in dx_execute.
+# We do not clone the current /var/delphix right away as the clone will be
+# outdated by the time dx_execute is called. Before deleting the dataset we
+# first verify that it doesn't contain anything useful.
+#
+(
+	cd "$TMP_ROOT/var/delphix" || die "failed to cd into $TMP_ROOT/var/delphix"
+	# We only expect to find:
+	#  ./dropbox
+	[[ $(find . -mindepth 1 | wc -l) -eq 1 ]] ||
+		die "linux dataset for /var/delphix contains unexpected files"
```

Comment:
these `die` statements will only exit the subshell, and not the parent shell, right? so, I think we should do something like:
```
(
    ...
) || die "verification of /var/delphix failed"
```

Reply:
Yeah, you are perfectly right... Nice catch, almost got tricked by a subshell issue again.